Hide home page redirect javascript URLs

keianhzo opened the pull request:
Fixes #617  Hide all javascript: urls from the URL bar

cvan commented on app/src/common/shared/org/mozilla/vrbrowser/ui/NavigationURLBar.java:
```diff
@@ -135,6 +135,8 @@ else if (aURL.startsWith("resource:") || SessionStore.get().isHomeUri(aURL))
                 aURL = "";
             else if (aURL.startsWith("data:") && SessionStore.get().isCurrentSessionPrivate())
                 aURL = "";
+            else if (aURL.startsWith("javascript:"))
```
nah, we should not always hide them. if a user enters a URL like this or gets redirected/clicked, it should show in the URL bar. can we hide it for only when the Home page is loaded?

keianhzo replied:
Fixed

cvan commented on app/src/common/shared/org/mozilla/vrbrowser/ui/NavigationURLBar.java:
```diff
@@ -135,7 +135,7 @@ else if (aURL.startsWith("resource:") || SessionStore.get().isHomeUri(aURL))
             else if (aURL.startsWith("data:") && SessionStore.get().isCurrentSessionPrivate())
                 aURL = "";
-            else if (aURL.startsWith("javascript:"))
+            else if (aURL.startsWith("javascript:") && SessionStore.get().isHomeUri(aURL))
```
not sure how this could work because `aURL` cannot start with `javascript:` and also start with `https://webxr.today/`

keianhzo commented:
@cvan Should be fine now. For some reason I couldn't see the "javascript:" string with the previous patch.

cvan commented on app/src/common/shared/org/mozilla/vrbrowser/ui/NavigationURLBar.java:
```diff
@@ -135,6 +135,8 @@ else if (aURL.startsWith("resource:") || SessionStore.get().isHomeUri(aURL))
                 aURL = "";
             else if (aURL.startsWith("data:") && SessionStore.get().isCurrentSessionPrivate())
                 aURL = "";
+            else if (aURL.startsWith("javascript:") && aURL.contains(SessionStore.get().getHomeUri()))
```
this fixes the bug, but it could trigger false positives. can we check that the previous URL started with the Home URI (`SessionStore.HOME_WITHOUT_REGION_ORIGIN`), and, if so, any URL changes or redirect (e.g., [`"javascript:window.location.replace('" + getHomeUri() + "')"`](https://github.com/MozillaReality/FirefoxReality/blob/81cbd41/app/src/common/shared/org/mozilla/vrbrowser/browser/SessionStore.java#L457)) gets correctly masked (i.e., `aURL remains ""`).

is there a way to set a flag for the active session before the [`loadUri` Gecko call here](https://github.com/MozillaReality/FirefoxReality/blob/81cbd41/app/src/common/shared/org/mozilla/vrbrowser/browser/SessionStore.java#L457) that will tell us "keep the URL empty when we are navigating on the Home page"?

keianhzo replied:
> is there a way to set a flag for the active session before the loadUri Gecko call here that will tell us "keep the URL empty when we are navigating on the Home page"?

There is no way of doing that in Gecko ATM.

> this fixes the bug, but it could trigger false positives. can we check that the previous URL started with the Home URI (SessionStore.HOME_WITHOUT_REGION_ORIGIN), and, if so, any URL changes or redirect (e.g., "javascript:window.location.replace('" + getHomeUri() + "')") gets correctly masked (i.e., aURL remains "").

Which are the false positives? Can you give me some example to reproduce? That will only leave the URL empty if it's any javascript URL that includes the home URL.

cvan commented:
also, can you change the commit/PR message to reflect what's being changed here (i.e., this is specific to hiding the `javascript:window.location.replace(…)` redirect for the Home/Start URI

keianhzo commented:
Fixed.

keianhzo commented:
@cvan Thanks for the URL I've added a getPreviousUri to deal with this. Also the URl have helped unveil a crash in Telemetry so I've also added a fix for that.
